optimizely/event/dispatcher.go

```diff
@@ -20,11 +23,26 @@ type Dispatcher interface {
 type HTTPEventDispatcher struct {
 }
 
-var dispatcherLogger = logging.GetLogger("EventDispatcher")
-
 // DispatchEvent dispatches event with callback
 func (*HTTPEventDispatcher) DispatchEvent(event LogEvent, callback func(success bool)) {
+	dispatchEvent(event, callback)
+}
+
+type QueueEventDispatcher struct {
+	q Queue
+	mux sync.Mutex
+}
+
+func (ed *QueueEventDispatcher) DispatchEvent(event LogEvent, callback func(success bool)) {
+
+	ed.q.Add(event)
+	callback(true)
```

Comment:
for the queued dispatch event, I have it calling the callback once it is queued.  the idea being that eventually, it would be sent and it is no longer the responsibility of the event processor.